src/udev/udev-manager.c

```diff
@@ -914,28 +914,22 @@ static int on_ctrl_msg(UdevCtrl *uctrl, UdevCtrlMessageType type, const UdevCtrl
                 }
 
                 eq++;
-                if (isempty(eq)) {
+                if (isempty(eq))
                         log_debug("Received udev control message (ENV), unsetting '%s'", key);
-
-                        r = hashmap_put(manager->properties, key, NULL);
-                        if (r < 0) {
-                                log_oom();
-                                return 1;
-                        }
-                } else {
+                else {
                         val = strdup(eq);
                         if (!val) {
                                 log_oom();
                                 return 1;
                         }
 
                         log_debug("Received udev control message (ENV), setting '%s=%s'", key, val);
+                }
 
-                        r = hashmap_put(manager->properties, key, val);
-                        if (r < 0) {
-                                log_oom();
-                                return 1;
-                        }
+                r = hashmap_put(manager->properties, key, val);
```

Comment:
Hmm... Not your fault, but this is a bug, we should remove the entry when `val` is NULL, and otherwise replace the existing entry.

Reply:
I assumed there is some weird reason there why the entry isn't really removed...

Reply:
https://github.com/systemd/systemd/pull/29931